src/common/meta/src/ddl/drop_database/metadata.rs

```diff
@@ -39,6 +41,7 @@ impl State for DropDatabaseRemoveMetadata {
             .schema_manager()
             .delete(SchemaNameKey::new(&ctx.catalog, &ctx.schema))
             .await?;
+        self.invalidate_schema_cache(ddl_ctx, ctx).await?;
```

Comment:
Better move this "cache invalidation" into a new step, like the `DropTableProcedure` does. So that cache invalidation's failure won't interfere with deleting metadata, good for procedure retry.

Reply:
@MichaelScofield refactor it. Add a new DropMetadataBroadcast State trait Impl.